### PR TITLE
[lexical-react] Bug Fix: Prevent false onChange event on initial autofocus with code highlighting

### LINKED ISSUE #4493: Bug: When AutoFocusPlugin is Combined with CodeHighlightPlugin and LexicalOnChangePlugin - an OnChange Event is raised on document load.
When the React AutoFocusPlugin,  (@lexical/react/LexicalAutoFocusPlugin) React OnChangePlugin (@lexical/react/LexicalOnChangePlugin) and the CodeHighlightPlugin (calling {registerCodeHighlighting} from '@lexical/code') are all enabled in the editor - then when the application first loads the document / editor and is 'AutoFocused' - an OnChange event is raised. 

Pretty sure it's being caused by the code highlighter registration - but it only happens when the AutoFocus (or the call to editor.focus()) occurs (i.e. it doesn't happen when the editor is manually focused by clicking inside the editor window).

Lexical version: 0.10.0

## Steps To Reproduce

1. Enable all three plugins - AutoFocusPlugin, OnChangePlugin, and CodeHighlightPlugin
2. Close the document / editor application and then re-open and existing document / editor - the editor will auto-focus, but it will also raise an onChange event - even though the editor state has not changed.

## The current behavior

An onChange event is being raised even though there is no change to editorState

## The expected behavior

There should be no onChange event when the document / editor is loaded (for example when a user is simply opening a document in the application)

### PR DESCRIPTION
**Pull request by matingathani**
## Description

Currently, when `AutoFocusPlugin` is combined with `CodeHighlightPlugin` and `LexicalOnChangePlugin`, the initial autofocus triggers an editor state change (due to code highlighting being applied), which causes `OnChangePlugin` to fire a false onChange event during the editor initialization.

This PR fixes the issue by capturing the initial editor state in a ref and filtering it out from the onChange callback, preventing the spurious event from being fired on document load.

Closes #4493

### Before

1. Use `AutoFocusPlugin` + `CodeHighlightPlugin` + `LexicalOnChangePlugin` together
2. Load the page -- onChange fires with an empty/dirty state on page load

### After

1. Same setup -- onChange no longer fires on page load
2. Normal user edits continue to trigger onChange as expected

### CODE REVIEW

**Review by etrepum — CHANGES_REQUESTED** [review]
Missing tests

**Review by etrepum — COMMENTED** [review]
I think there are some edge cases here when using something like the HistoryPlugin which can change the EditorState to a previously observed EditorState, but then the onChange wouldn't be called if that state transition happened to go back to the first observed state.
(outside of the update listener) to track this probably makes more sense.

**Comment by matingathani** [comment]
Thanks for the review @etrepum — you're right on both counts.

**HistoryPlugin edge case (fixed):** The ref-based suppression was applied on _every_ update, so after a HistoryPlugin undo all the way back to the initial state, any subsequent user action would have `prevEditorState === initialPrevEditorStateRef.current` and be incorrectly suppressed. Fixed by adding a `hasSeenFirstChangeRef` boolean: the initial-state comparison only applies _before_ the first real change has been seen. Once `onChange` has fired at least once, the guard is disabled for all future updates.

**Tests:** The existing test file covers:
1. Initial autofocus + SimulateFocusUpdatePlugin (code-highlight-style update from mount state) → `onChange` not called ✓  
2. A real user change after mount → `onChange` called once ✓  
3. **New test:** Mounts with SimulateFocusUpdatePlugin, makes a real change (`onChange` fires), then calls `editor.setEditorState(initialEditorState)` to simulate HistoryPlugin restoring to the initial state — `onChange` fires again (2×). Then a further user action from that restored state also fires (3×). Without the `hasSeenFirstChangeRef` guard, both the undo-restore and the subsequent action would have been incorrectly suppressed.

Let me know if you'd like a different approach for the edge-case guard.

File: packages/lexical-react/src/LexicalOnChangePlugin.ts [hunk 1]
```diff
@@ -26,21 +27,27 @@ export function OnChangePlugin({
   ) => void;
 }): null {
   const [editor] = useLexicalComposerContext();
+  const initialPrevEditorStateRef = useRef<EditorState | null>(null);
+  const hasSeenFirstChangeRef = useRef(false);
```
Comment (etrepum) [inline packages/lexical-react/src/LexicalOnChangePlugin.ts:31]: I don't think these need to be refs at all, you should be able to just use `let` variables in the effect outside of the update listener. There's no reason to preserve the state over multiple mounts of the effect
